test/test_provisioning_api.py

```diff
@@ -93,25 +103,64 @@ def test_update_user(self):
         self.assertEqual(new_email_address, res["email"])
 
         res = cloudinary.provisioning.users()
         user_by_id = [user for user in res["users"]
-                      if user["id"] == self.user_id]
+                      if user["id"] == self.user_id_1]
         self.assertEqual(len(user_by_id), 1)
 
     @unittest.skipUnless(cloudinary.provisioning.account_config().provisioning_api_secret,
                          "requires provisioning_api_key/provisioning_api_secret")
     def test_get_users(self):
-        res = cloudinary.provisioning.users(user_ids=[self.user_id])
+        res = cloudinary.provisioning.users(user_ids=[self.user_id_1])
+        self.assertEqual(len(res["users"]), 1)
+
+    @unittest.skipUnless(cloudinary.provisioning.account_config().provisioning_api_secret,
+                         "requires provisioning_api_key/provisioning_api_secret")
+    def test_get_pending_users(self):
+        res = cloudinary.provisioning.users(user_ids=[self.user_id_1], pending=True)
         self.assertEqual(len(res["users"]), 1)
 
+    @unittest.skipUnless(cloudinary.provisioning.account_config().provisioning_api_secret,
+                         "requires provisioning_api_key/provisioning_api_secret")
+    def test_get_non_pending_users(self):
+        res = cloudinary.provisioning.users(user_ids=[self.user_id_1], pending=False)
+        self.assertEqual(len(res["users"]), 0)
+
+    @unittest.skipUnless(cloudinary.provisioning.account_config().provisioning_api_secret,
+                         "requires provisioning_api_key/provisioning_api_secret")
+    def test_get_pending_and_non_pending_users(self):
+        res = cloudinary.provisioning.users(user_ids=[self.user_id_1], pending=None)
+        self.assertEqual(len(res["users"]), 1)
+
+    @unittest.skipUnless(cloudinary.provisioning.account_config().provisioning_api_secret,
+                         "requires provisioning_api_key/provisioning_api_secret")
+    def test_get_users_by_prefix(self):
+        res_1 = cloudinary.provisioning.users(pending=True, prefix=self.user_name_2[:-1])
+        res_2 = cloudinary.provisioning.users(pending=True, prefix=self.user_name_2+'zzz')
+        self.assertEqual(len(res_1["users"]), 1)
+        self.assertEqual(len(res_2["users"]), 0)
+
+    @unittest.skipUnless(cloudinary.provisioning.account_config().provisioning_api_secret,
+                         "requires provisioning_api_key/provisioning_api_secret")
+    def test_get_users_by_sub_account_id(self):
+        res = cloudinary.provisioning.users(pending=True, user_ids=[self.user_id_2], sub_account_id=self.cloud_id)
+        self.assertEqual(len(res["users"]), 1)
+
+    @unittest.skipUnless(cloudinary.provisioning.account_config().provisioning_api_secret,
+                         "requires provisioning_api_key/provisioning_api_secret")
+    def test_get_users_by_nonexistent_sub_account_id(self):
+        random_id = random.randint(100000, 999999)
```

Comment:
Let's use something similar to `UNIQUE_TEST_ID` or `UNIQUE_TAG` from `helper_test.py` instead of just calling random.